internal/filesystem.go

```diff
@@ -107,6 +108,43 @@ func (rdl *runDirListener) Addr() net.Addr {
 // ManualListener returns manually created listener with provided permissions
 func ManualListener(socket string, perm fs.FileMode) func() (net.Listener, error) {
 	return func() (net.Listener, error) {
+		pidfile := DaemonPid
+		// check and read pid file
+		if FileExists(pidfile) {
+			out, err := FileRead(pidfile)
+			if err != nil {
+				// pid file exists, but is not readable,
+				// some garbage from previous run/failure? maybe we can cleanup and proceed?
+				return nil, fmt.Errorf("daemon pid file is not readable: %w", err)
```

Comment:
If the PID file exists, but fails to read, does it mean that the daemon will not start?

Reply:
yes, but as we under `root` it would be some kind of very extreme situation.